src/badguy/dart.hpp

```diff
@@ -49,10 +52,15 @@ class Dart final : public BadGuy
   virtual void stop_looping_sounds() override;
   virtual void play_looping_sounds() override;
 
+  virtual void on_flip(float height) override;
+
 protected:
   const BadGuy* parent; /**< collisions with this BadGuy will be ignored */
   std::unique_ptr<SoundSource> sound_source; /**< SoundSource for ambient sound */
 
+private:
```

Comment:
I believe you can move the `m_flip` variable into a base class that all objects inherit from. Either as a public variable or with getters and setters.